src/create-reminder-signup/lambda/lambda.ts

```diff
@@ -37,7 +37,8 @@ export const run = async (
 ): Promise<APIGatewayProxyResult> => {
 	console.log('received event: ', event);
 
-	const signupRequest: unknown = JSON.parse(event.body);
+	const country = event.headers['X-GU-GeoIP-Country-Code'];
+	const signupRequest: unknown = { ...JSON.parse(event.body), country };
```

Comment:
This is technically unsafe as `JSON.parse` may not return an object. We could potentailly check for that 

```js
const body = JSON.parse(event.body);
if (body === null || typeof body !== "object") {
  /* return an error (400/422?) */
}
```

Reply:
Indeed, and JSON.parse can throw an exception if it fails!
```
> JSON.parse('*')
Uncaught SyntaxError: JSON.parse: unexpected character at line 1 column 1 of the JSON data
```

Reply:
Yeah - I'm just trying to dig through the docs now to see if amazon give any gaurantees about what it can be. They mention that it's a json representation of the http request. I think that means we can be confident it will be valid json. I don't know if it's possible to be a string that's anything other than an object though?

Reply:
<img width="641" alt="Screenshot 2021-02-09 at 09 31 34" src="https://user-images.githubusercontent.com/17720442/107343884-b9955e00-6ab9-11eb-9437-3b8777df5fd3.png">

Reply:
It can! I'll add some checks!